### PR TITLE
Implement GET/POST for shard chains

### PR DESCRIPTION
**Pull request by SamWilsn**

### CODE REVIEW

File: notion-server/src/api/mod.rs [hunk 1]
```diff
@@ -78,3 +80,29 @@ async fn create_execution_environment(
...
     Ok(status::Created(location.to_string(), None))
 }
+
+#[tokio::main]
+#[post("/shards", data = "<shard>")]
+async fn create_shard_chain(
+    shard: Json<args::CreateShardChain>,
+    handle: State<Handle>,
+) -> EthResult<status::Created<()>> {
+    let shard = shard.into_inner();
```
Comment (gjtrowbridge) [inline notion-server/src/api/mod.rs:90]: I assume this just converts the `Json` object into the internal `CreateShardChain` arg?

File: notion-server/src/api/mod.rs [hunk 2]
```diff
@@ -78,3 +80,29 @@ async fn create_execution_environment(
...
+
+    let idx = handle.clone().create_shard_chain(shard).await?;
+    let location = uri!(show_shard_chain: idx);
```
Comment (gjtrowbridge) [inline notion-server/src/api/mod.rs:93]: Clever, I like it.

**Review by gjtrowbridge — APPROVED** [review]
LGTM

File: notion-server/src/api/mod.rs (hunk 1 above)
Comment (SamWilsn) [reply notion-server/src/api/mod.rs:90]: Yep! The `Json` struct wraps the `CreateShardChain` struct to give Rocket a hint about how to deserialize it. There are other wrappers for other encodings.

File: notion-server/src/api/mod.rs (hunk 2 above)
Comment (SamWilsn) [reply notion-server/src/api/mod.rs:93]: Rocket is magic sometimes.

**Pull request merged**